### PR TITLE
Enable NetCore scenario tests and set up scenario test build

### PR DESCRIPTION
**Pull request by markcowl**
## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/EnvironmentSetupHelper.cs [hunk 1]
```diff
@@ -53,8 +54,15 @@ public class EnvironmentSetupHelper
         private AzureAccount testAccount;
 
         private const string PackageDirectoryFromCommon = @"..\..\..\..\Package\Debug";
-        public string PackageDirectory = @"..\..\..\..\..\Package\Debug";
-        public string StackDirectory = @"..\..\..\..\..\..\Stack\Debug";
+
+        public static string PackageDirectory { get; }  = GetConfigDirectory();
```
Comment (MiYanni) [inline src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/EnvironmentSetupHelper.cs]: Good job on the directory cleanup. Will help when we restructure.

File: src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/Common.ResourceManager.ScenarioTests.Netcore.csproj [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 
-  <Import Project="..\..\..\..\tools\Common.Netcore.Dependencies.targets" />
-
   <PropertyGroup>
     <TargetFramework>netcoreapp2.0</TargetFramework>
     <AssemblyName>Microsoft.Azure.Commands.ScenarioTest.Common</AssemblyName>
     <RootNamespace>Microsoft.Azure.Commands.ScenarioTest</RootNamespace>
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+    <OutputType>Exe</OutputType>
+    <RuntimeIdentifiers>win10-x64;linux-x64;osx.10.12-x64</RuntimeIdentifiers>
```
Comment (MiYanni) [inline src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/Common.ResourceManager.ScenarioTests.Netcore.csproj]: [RuntimeIdentifiers](https://docs.microsoft.com/en-us/dotnet/core/tools/csproj#runtimeidentifiers) are only used for [publishing](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-publish?tabs=netcore2x), and should not be required.

File: src/ResourceManager/Profile/Commands.Profile.Test/Commands.Profile.Test.Netcore.csproj [hunk 3]
```diff
@@ -1,12 +1,10 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
-
-  <Import Project="..\..\..\..\tools\Common.Netcore.Dependencies.targets" />
-
   <PropertyGroup>
     <TargetFramework>netcoreapp2.0</TargetFramework>
     <AssemblyName>Microsoft.Azure.Commands.Profile.Test</AssemblyName>
     <RootNamespace>Microsoft.Azure.Commands.Profile.Test</RootNamespace>
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+    <RuntimeIdentifiers>win10-x64;linux-x64;osx.10.12-x64</RuntimeIdentifiers>
```
Comment (MiYanni) [inline src/ResourceManager/Profile/Commands.Profile.Test/Commands.Profile.Test.Netcore.csproj]: [RuntimeIdentifiers](https://docs.microsoft.com/en-us/dotnet/core/tools/csproj#runtimeidentifiers) are only used for [publishing](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-publish?tabs=netcore2x), and should not be required.

File: src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/Common.ResourceManager.ScenarioTests.Netcore.csproj [hunk 4]
```diff
@@ -1,12 +1,12 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
-  <Import Project="..\..\..\..\tools\Common.Netcore.Dependencies.targets" />
-
   <PropertyGroup>
     <TargetFramework>netcoreapp2.0</TargetFramework>
     <AssemblyName>Microsoft.Azure.Commands.ScenarioTest.Common</AssemblyName>
     <RootNamespace>Microsoft.Azure.Commands.ScenarioTest</RootNamespace>
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+    <OutputType>Exe</OutputType>
```
Comment (MiYanni) [inline src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/Common.ResourceManager.ScenarioTests.Netcore.csproj]: Does this need to be an exe?

File: src/ResourceManager/Profile/Commands.Profile.Test/Commands.Profile.Test.Netcore.csproj [hunk 5]
```diff
@@ -22,48 +20,51 @@
   </PropertyGroup>
 
   <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp2.0' ">
```
Comment (MiYanni) [inline src/ResourceManager/Profile/Commands.Profile.Test/Commands.Profile.Test.Netcore.csproj]: I'm not sure this is necessary (I know you didn't add it). Just noting it.

File: tools/Common.Netcore.Dependencies.targets [hunk 6]
```diff
@@ -2,8 +2,11 @@
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp2.0' ">
     <PackageReference Include="Microsoft.ApplicationInsights" Version="2.4.0" />
-    <PackageReference Include="Microsoft.Azure.Common" Version="2.1.4" />
-    <PackageReference Include="PowerShellStandard.Library" Version="3.0.0-preview-01" />
+    <PackageReference Include="Microsoft.Azure.Common" Version="2.2.0" />
```
Comment (MiYanni) [inline tools/Common.Netcore.Dependencies.targets]: Would these be the same dependencies that should exist in Common.Netcore.Dependencies.Test.targets? (file doesn't exist yet, but will in due time)

File: src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/EnvironmentSetupHelper.cs [hunk 7]
```diff
@@ -101,109 +110,79 @@ public EnvironmentSetupHelper()
...
-            {
-                return Path.Combine(this.PackageDirectory,
-                                    @"ResourceManager\AzureResourceManager\AzureRM.Storage\AzureRM.Storage.psd1");
-            }
-        }
+        public string RMStorageModule { get; } = GetModuleManifest(RmDirectory, "AzureRM.Storage");
 
         //TODO: clarify (data plane should not be under ARM folder)
-        public string RMStorageDataPlaneModule
+        public string RMStorageDataPlaneModule { get; } = GetModuleManifest(StorageDirectory, "AzureR.Storage");
```
[the remark this answers is not in the document]
Comment (markcowl) [reply src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/EnvironmentSetupHelper.cs]: Actually, should be Azure.Storage, good catch

File: src/ResourceManager/Automation/Commands.Automation/Common/RequestSettings.cs [hunk 8]
```diff
@@ -31,7 +30,6 @@ public RequestSettings(IAutomationManagementClient automationClient)
 
             client.HttpClient.DefaultRequestHeaders.Remove(Constants.ActivityIdHeaderName);
             var activityId = Guid.NewGuid();
-            EventProvider.SetActivityId(ref activityId);
```
[the remark this answers is not in the document]
Comment (markcowl) [reply src/ResourceManager/Automation/Commands.Automation/Common/RequestSettings.cs]: Yes, this functionality is not available in NetCore

File: src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/Common.ResourceManager.ScenarioTests.Netcore.csproj (hunk 2 above)
Comment (markcowl) [reply src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/Common.ResourceManager.ScenarioTests.Netcore.csproj]: Yes, I had forgotten about the experiment to package as an exe and run tests from the package.  Will remove these.

File: tools/Common.Netcore.Dependencies.targets (hunk 6 above)
Comment (markcowl) [reply tools/Common.Netcore.Dependencies.targets]: So, I think the netcore common test should include the test libraries and PowerShell SDK.  Once we have two or three converted we can add additional libraries to common

**Comment by markcowl** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/745/

signing run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/ps-sign/436/

netsign run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/psnetsign/59/

**Comment by markcowl** [comment]
new ps netsign after azurerm.netcore update: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/psnetsign/60/

**Pull request merged**